### DISCUSSION TITLE
Provide spdlog with a `Clock` object

### DISCUSSION THREAD

**Discussion by michael-projectx** [gabime/spdlog#2887] [Ideas]
I'm currently using `spdlog` in my project which runs both on hardware and simulation. In simulation we have a simulated clock that is not in sync with the system clock. When using `spdlog` in the context of our simulator, the timestamps printed in the spdlog messages are meaningless. Is there a way to provide `spdlog` with a specific clock?

**Comment by tt4g** [comment]
If you have replaced the clock, change the next line.

https://github.com/gabime/spdlog/blob/91807c2e718890ca2c0c88620326c6919ce041f4/include/spdlog/common.h#L132

Or pass a `time_point` object directly to the `spdlog::logger::log()` function.

https://github.com/gabime/spdlog/blob/91807c2e718890ca2c0c88620326c6919ce041f4/include/spdlog/logger.h#L101-L104
